tests/agent_unittests/test_harvest_loop.py

```diff
@@ -537,13 +537,13 @@ def test_reservoir_sizes(transaction_node):
 
 
 @pytest.mark.parametrize(
-    "harvest_name, event_name",
+    "harvest_name, harvest_setting, event_name",
     [
-        ("analytic_event_data", "transaction_events"),
-        ("error_event_data", "error_events"),
-        ("custom_event_data", "custom_events"),
-        ("log_event_data", "log_events"),
-        ("span_event_data", "span_events"),
+        ("max_samples_stored", "transaction_events", "transaction_events"),
+        ("max_event_samples_stored", "error_collector", "error_events"),
+        ("max_samples_stored", "custom_insights_events", "custom_events"),
+        ("max_samples_stored", "application_logging.forwarding", "log_events"),
+        ("max_samples_stored", "span_events", "span_events"),
     ],
 )
```

Comment:
Let's make the setting names here more readable

```suggestion
@pytest.mark.parametrize(
    "harvest_setting,event_name",
    [
        ("transaction_events.max_samples_stored", "transaction_events"),
        ("error_collector.max_event_samples_stored", "error_events"),
        ("custom_insights_events.max_samples_stored", "custom_events"),
        ("application_logging.forwarding.max_samples_stored", "log_events"),
        ("span_events.max_samples_stored", "span_events"),
    ],
)
```